src/storage-client/src/sink.rs

```diff
@@ -430,105 +448,92 @@ async fn determine_latest_progress_record(
         // transactions, there might even be a lot of garbage at the end of the
         // topic or in between.
 
+        // First, determine the current high water mark for the progress topic.
+        // This is the position our `progress_client` consumer *must* reach
+        // before we can conclude that we've seen the latest progress record for
+        // the specified `progress_key`. A safety argument:
+        //
+        //   * The producer has initialized transactions before calling this
+        //     method. At this point, any writes from a previous version of this
+        //     sink should be visible, so the offset we fetch should be larger
```

Comment:
This comment in this particular state of the code is not true because here we'll get the LSO. Maybe leave a TODO that Sean's PR removes?

Reply:
Good call. Done.